### PR TITLE
[CHIP] Add support fox Xcode 13 beta 3.

### PR DESCRIPTION
**Pull request by mandel-macaque**

### CODE REVIEW

**Review by rolfbjarne — APPROVED** [review]

File: src/chip.cs [hunk 1]
```diff
@@ -1123,6 +1134,14 @@ interface ChipOperationalCredentials
 		[DesignatedInitializer]
 		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
 
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("addOpCert:iCACertificate:iPKValue:caseAdminNode:adminVendorId:responseHandler:")]
+		void AddOpCert (NSData noc, NSData iCACertificate, NSData iPKValue, ulong caseAdminNode, ushort adminVendorId, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("opCSRRequest:responseHandler:")]
+		void OpCSRRequest (NSData cSRNonce, ChipResponseHandler responseHandler);
```
Comment (spouliot) [inline src/chip.cs:1143]: `OpCsrRequest` -> `Csr` -> TLA

File: src/chip.cs [hunk 2]
```diff
@@ -1764,4 +1799,706 @@ interface ChipQRCodeSetupPayloadParser
...
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("getSetupPIN:responseHandler:")]
+		void GetSetupPin (string tempAccountIdentifier, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("login:setupPIN:responseHandler:")]
+		void Login (string tempAccountIdentifier, string setupPIN, ChipResponseHandler  responseHandler);
```
Comment (spouliot) [inline src/chip.cs:1818]: -> `setupPin`

File: src/chip.cs (hunk 1 above)
Comment (spouliot) [reply src/chip.cs:1143]: parameter name should be `csrNonce` too

**Review by spouliot — CHANGES_REQUESTED** [review]

File: src/chip.cs [hunk 3]
```diff
@@ -1140,7 +1140,7 @@ interface ChipOperationalCredentials
 
 		[Async (ResultTypeName = "ChipReadAttributeResult")]
 		[Export ("opCSRRequest:responseHandler:")]
-		void OpCSRRequest (NSData cSRNonce, ChipResponseHandler responseHandler);
+		void OpCsRrequest (NSData csrNonce, ChipResponseHandler responseHandler);
```
Comment (spouliot) [inline src/chip.cs:1143]: wrong `r`
`OpCsRrequest` -> `OpCsrRequest`

**Review by rachelkang — COMMENTED** [review]
The diff shows that the NSError param in the CHIPDevicePairingDelegate methods onPairingComplete, onPairingDeleted, and onAddressUpdated are now nullable. Should there be a [NullAllowed] attribute added to those?

Similarly, should CHIPGetKeyValue be decorated with a [NullAllowed] attribute as well?

File: src/chip.cs [hunk 4]
```diff
@@ -1764,4 +1799,706 @@ interface ChipQRCodeSetupPayloadParser
...
+		[Export ("windowCoveringGoToTiltValue:responseHandler:")]
+		void WindowCoveringGoToTiltValue (ushort tiltValue, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("windowCoveringStop:")]
+		void WindowCoveringStop (ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("windowCoveringUpOpen:")]
+		void WindowCoveringUpOpen (ChipResponseHandler responseHandler);
```
Comment (rachelkang) [inline src/chip.cs:2424]: I think you can (maybe?) get rid of the `WindowCovering` prefix on all of the above

File: src/chip.cs (hunk 3 above)
Comment (mandel-macaque) [inline src/chip.cs:1143]:
```suggestion
		void OpCsrRequest (NSData csrNonce, ChipResponseHandler responseHandler);
```

**Review by spouliot — APPROVED** [review]
nice catches @rachelkang !

File: src/chip.cs [hunk 5]
```diff
@@ -50,6 +50,16 @@ public enum ChipRendezvousInformationFlags : ulong {
...
+	[Mac (12,0), Watch (8,0), TV (15,0), iOS (15,0), MacCatalyst (15,0)]
+	[Native]
+	public enum ChipCommissioningFlow : ulong {
+		Standard = 0,
+		UserActionRequired = 1,
+		Custom = 2,
+		Invalid = 3,
+	}
+
+
```
Comment (tj-devel709) [inline src/chip.cs:62]: small: extra newline

File: src/chip.cs [hunk 6]
```diff
@@ -1764,4 +1800,706 @@ interface ChipQRCodeSetupPayloadParser
...
+		[DesignatedInitializer]
+		IntPtr Constructor (ChipDevice device, byte endpoint, DispatchQueue queue);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("getSetupPIN:responseHandler:")]
+		void GetSetupPin (string tempAccountIdentifier, ChipResponseHandler responseHandler);
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("login:setupPIN:responseHandler:")]
+		void Login (string tempAccountIdentifier, string setupPin, ChipResponseHandler  responseHandler);
```
Comment (tj-devel709) [inline src/chip.cs:1819]:
```suggestion
		void Login (string tempAccountIdentifier, string setupPin, ChipResponseHandler responseHandler);
```
small: extra space here and next selector

File: src/chip.cs [hunk 7]
```diff
@@ -1764,4 +1800,706 @@ interface ChipQRCodeSetupPayloadParser
...
+
+		[Async (ResultTypeName = "ChipReadAttributeResult")]
+		[Export ("readAttributeClusterRevisionWithResponseHandler:")]
+		void ReadAttributeClusterRevision (ChipResponseHandler  responseHandler);
```
Comment (tj-devel709) [inline src/chip.cs:1823]:
```suggestion
		void ReadAttributeClusterRevision (ChipResponseHandler responseHandler);
```

**Review by tj-devel709 — APPROVED** [review]

File: src/chip.cs (hunk 5 above)
Comment (mandel-macaque) [inline src/chip.cs:62]:
```suggestion

```

**Pull request merged**